dbt_project.yml

```diff
@@ -0,0 +1,43 @@
+name: "dbt_mapzap"
+version: "1.0.0"
+config-version: 2
+
+profile: "dbt_mapzap"
+
+model-paths: ["mapzap/models"]
+seed-paths: ["mapzap/seeds"]
+test-paths: ["mapzap/tests"]
+docs-paths: ["docs"]
+
+clean-targets:
+  - "target"
+
+seeds:
+  +quote_columns: false
+
+models:
+  dbt_mapzap:
+    staging:
+      +materialized: view
```

Comment:
So I was confused by their language here so I did some reading just to make sure, and want to clarify intended usage here.

I was confused because in postgres, there are materialized and non-materialized views, the former of which is stored (as a table essentially) and the latter which is just essentially a stored query. dbt uses materialized just to specify what sort of output there is, and the view here is NOT materialized. Which could be fine - less data actually being stored during running. But depending on downstream queries (and if a model needs to be queried multiple times) it could be better to store as a table

Reply:
that sounds right, I definitely didn't learn enough about the pros/cons to optimize the choices between view and table here

going off of their general recommendations, the only non-view setting I made was for the `product` model, since it's the "big final table" and I imagined many queries on it to see and check the final results

Reply:
Sounds good. I think as a rule defaulting to views until you run into performance issues makes sense (or if you know you have one model that's going to be referenced a lot within the build for some reason)